File: ruby/roman-numerals/roman_numerals.rb
class Integer
    ROMAN =  {
        1000 => "M",  
        900  => "CM",  
        500  => "D",  
        400  => "CD",
        100  => "C",  
        90   => "XC",  
        50   => "L",  
        40   => "XL",  
        10   => "X",  
        9    => "IX",  
        5    => "V",  
        4    => "IV",  
        1    => "I"  
      }
    def to_roman
        arabic_number= self
        roman_number= ""

        ROMAN.each do |number, letter|
            roman_number << letter * (arabic_number / number)
            arabic_number = arabic_number % number
        end
        roman_number
    end
end

File: javascript/roman-numerals/roman-numerals.js
const roman_arabic = {
  M: 1000,
  CM: 900,
  D: 500,
  CD: 400,
  C: 100,
  XC: 90,
  L: 50,
  XL: 40,
  X: 10,
  IX: 9,
  V: 5,
  IV: 4,
  I: 1
};

export const toRoman = number => {
  let result = "";
  for (const roman in roman_arabic) {
    const arabic = roman_arabic[roman];
    
    while (number >= arabic) {
      result += roman;
      number -= arabic;
    }
  }
  return result;
};
